python/src/iceberg/expressions/base.py

```diff
@@ -267,6 +228,34 @@ def __str__(self) -> str:
         return "false"
 
 
+class UnboundIn(BooleanExpression, UnboundPredicate):
```

Comment:
To implement `UnboundPredicate`, this needs a `bind` method that returns an `In` instance.

Reply:
`UnboundPredicate` can be abstract and define `bind`.